### PR TITLE
Introduce new way to provide function execution

### PR DESCRIPTION
**Pull request by ceelsoin**
This is a work in progress

We are introducing a new way to implement function execution wrapper to another platforms. This will able users to create and execution serverless functions in Go, Python, node and more.

Our proposal:
image

Notes: 
- All code deployment and management will be made inside the platform. 
- Native execution way using `@globocom/functions-sandbox` package will not break

Our roadmap:
- Manage function execution in other providers
- Execution functions over GCF
- Deploy code to GCF through functions
- Manage function environment vars through functions

### CODE REVIEW

**Review by marcelometal — CHANGES_REQUESTED** [review]
Please, use config variables for all google params

File: lib/domain/GoogleFunctionProvider.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      return `Bearer ${token}`
+    }catch(err){
+      throw err
+    }
+  }
+
+  async run(req) {
+    try {
+      const zoneId = "us-central1"
+      const projectId = "gglobo-bs-events-hdg-dev"
```
Comment (marcelometal) [inline lib/domain/GoogleFunctionProvider.js:27]: Ooooops!

IMHO, Keep your private information private

Comment (ceelsoin) [reply lib/domain/GoogleFunctionProvider.js]: lol, not really dangerous, but i changed to env vars, we are making decisions about these ids will be defined by env or function definition endpoint

File: lib/domain/GoogleFunctionProvider.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "Content-Type": "application/json",
+        "Authorization": token
+      }
+    })
+
+    return response
+  }
+
+  async deployFunction(){
+    const projectId = "gglobo-bs-events-hdg-dev"
```
Comment (marcelometal) [inline lib/domain/GoogleFunctionProvider.js]: Ooooops!

IMHO, Keep your private information private

**Pull request closed without merging**